### ISSUE TITLE
FYI: How to install man pages for X (easy)

### ISSUE THREAD

**Issue by coolaj86** [webinstall/webi-installers#338]
# TL;DR

`MANPATH` is generated from `PATH` - just backing up from `./bin` and going to `./share/man`.

All installers should install their man pages to either:

```txt
~/.local/share/man
```

or

```txt
~/.local/opt/${PKG_NAME}/share/man
```

# Explanation

The `manpath` program will tell you what's in your MANPATH.

In fact, MANPATH automatically includes each item from PATH, modified to search for `../share/man`.

For example:

If `~/.local/bin` is in my path, then `~/.local/share/man` is automatically in my MANPATH, if the folder exists.

# MANPATH derived from PATH

man pages should be installed relative to the package install location.

| If `yq` is installed at ... | then man pages should go in ... |
| ----------------------  | ------------------------------- |
| `~/bin/yq`                      | `~/share/man/man1/yq.1` |
| `~/.local/bin/yq`            | `~/.local/share/man/man1/yq.1` |
| `~/.local/opt/yq/bin/yq` | `~/.local/opt/yq/share/man/man1/yq.1` |
| `/usr/local/bin/yq` | `/usr/local/share/man/man1/yq.1` |

The pattern is "if `PATH=$foo/bin` then `MANPATH=$foo/share/man`. This is automatic. 

MANPATH is updated by virtue of PATH being updated.

# How Webi Works

Webi automatically adds `~/.local/bin` and `~/.local/opt/WHATEVER/bin` to the `PATH`, and so the `MANPATH` search will always include those man files as well, assuming they are installed.

**Comment by coolaj86** [comment]
# Example Script (webi)

```bash
    pkg_install() {
        mkdir -p "$(dirname "$pkg_src_cmd")"
        # yq_linux_amd64.tar.gz contains:
        #   - yq_linux_amd64
        #   - yq.1
        #   - install-man-page.sh
        if [[ -e ./yq.1 ]]; then
            mkdir -p ~/.local/share/man/man1
            mv ./yq.1 ~/.local/share/man/man1/
        fi
        mv ./"$pkg_cmd_name"* "$pkg_src_cmd"
        chmod a+x "$pkg_src_cmd"
    }
```

# Example Script (generic)

```bash
#!/bin/sh

my_path="$(command -v yq)"

if [ -z "$my_path" ]; then
  echo "'yq' wasn't found in your PATH, so we don't know where to put the man pages."
  echo "Please update your PATH to include yq, and run this script again."
  exit 1
fi

# ex: ~/.local/bin/yq => ~/.local/
my_prefix="$(dirname "$(dirname "$(command -v yq)")")"
mkdir -p "$my_prefix/share/man/man1/"
cp yq.1 "$my_prefix/share/man/man1/"
```
